packages/owl-bot/src/copy-code.ts

```diff
@@ -116,19 +156,52 @@ ${err}`,
   );
   cmd('git add -A', {cwd: destDir});
   cmd(`git commit -F "${commitMsgPath}" --allow-empty`, {cwd: destDir});
+  return {
+    kind: 'LocalCopy',
+    dir: destDir,
+    sourceCommitHash,
+  };
+}
+
+/**
+ * Copies the code from googleapis-gen to the dest repo, and creates a
+ * pull request.
+ * @param sourceRepo: the source repository, either a local path or googleapis/googleapis-gen
+ * @param sourceRepoCommit: the commit from which to copy code. Empty means the most recent commit.
+ * @param destRepo: the destination repository, either a local path or a github path like googleapis/nodejs-vision.
+ */
+export async function copyCodeAndCreatePullRequest(
+  sourceRepo: string,
+  sourceRepoCommitHash: string,
+  destRepo: GithubRepo,
+  octokitFactory: OctokitFactory,
+  logger = console
+): Promise<void> {
+  const destBranch = 'owl-bot-' + uuidv4();
+  const dest = await copyCodeIntoLocalBranch(
+    sourceRepo,
+    sourceRepoCommitHash,
+    destRepo,
+    destBranch,
+    octokitFactory,
+    logger
+  );
+  if (dest.kind === 'CreatedGithubIssue') {
```

Comment:
If I'm understanding correctly, `dest.kind` is used to create a state machine where we react to the responses of several steps in the pull request creation process that could have multiple outcomes, e.g., opening an issue rather than updating a PR?

Reply:
`.kind` is just used to distinguish the two possible return values.  I'm following the typescript convention described here: https://www.typescriptlang.org/docs/handbook/2/narrowing.html